### ISSUE TITLE
Update Stratum API method names to match bitcoin standard

### ISSUE THREAD

**Issue by bladedoyle** [mimblewimble/grin#1166]
As stated in the title this ticket tracks the change from monero-style RPC method and parameter names to bitcoin-style naming - as closely as possible.

Reference: https://en.bitcoin.it/wiki/Stratum_mining_protocol

**Comment by quentinlesceller** [comment]
Is this still an ongoing issue or we are good with the current implementation ?

**Comment by bladedoyle** [comment]
This is per @ignopeverell request.
I only put it off because its an (another) RPC protocol breaking change which will cause some disruption.
When I looked I didnt see much to change, but am open for suggestions.  I'll find some time to document proposed changes and request comments.

**Comment by bladedoyle** [comment]
###  Client to Server
| current | new |
|-----------|-------|
| login | mining.authorize |
| submit |  mining.submit |
| getjobtemplate | mining.subscribe |


### Server to client
| current | new |
|-----------|-------|
| job | mining.notify |

**Comment by ignopeverell** [comment]
Yes, that looks good to me, we have few methods so now is a perfect time to do this. Should we ever worry about supporting the (suggest/set)_difficutly methods? I remember you mentioning that before but forgot the conclusion.
